### PR TITLE
Lint tests: Directories O-Resource

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
More linting of test files.

### CODE REVIEW

File: tests/optimize/test_adagrad.py [hunk 1]
```diff
@@ -1,3 +1,19 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
```
Comment (timmysilv) [inline tests/optimize/test_adagrad.py:1]:
```suggestion
# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
```

**Review by timmysilv — COMMENTED** [review]
just one concern, but otherwise lgtm!! thank you for keeping this moving!

File: tests/optimize/test_spsa.py [hunk 2]
```diff
@@ -239,8 +239,9 @@ def quant_fun_mdarr(var):
                 multiplier = ck * delta
                 thetaplus[index] = arg + multiplier
                 thetaminus[index] = arg - multiplier
-            yplus = quant_fun_mdarr(*thetaplus)
-            yminus = quant_fun_mdarr(*thetaminus)
+            print(thetaplus)
```
Comment (mudit2812) [inline tests/optimize/test_spsa.py:242]:
```suggestion
```

**Review by mudit2812 — APPROVED** [review]
Thanks David! This looks good to me. One thing I've noticed is that there are some pylint errors that are disabled in many test files, such as `too-many-arguments`, `too-few-public-methods`, `too-many-public-methods`, `protected-access`. I think all of these errors should always be ignored, as they are completely irrelevant to code quality specifically in the context of testing.

It might be worth adding including these warnings in the ignore section of the `tests/.pylintrc` file in the next PR. It'll definitely make your life easier as well :)

File: tests/ops/functions/test_matrix.py [hunk 3]
```diff
@@ -606,7 +609,6 @@ def test_get_unitary_matrix_interface_jax(self):
         from jax import numpy as jnp
         from jax.config import config
 
-        remember = config.read("jax_enable_x64")
```
[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/ops/functions/test_matrix.py:609]: Haha, I encountered this line the first time for this PR and it is not used.. Maybe somethingsomething precision agnosticity :)

File: tests/ops/qubit/test_qchem_ops.py [hunk 4]
```diff
@@ -441,38 +424,8 @@ def test_double_excitation_decomp(self, phi):
...
-            for i in reversed(decomp):
-                # Single-qubit gate
-                if len(i.wires.tolist()) == 1:
-                    mat = single_mat_four_qubits(i.matrix(), i.wires.tolist()[0])
-                    mats.append(mat)
-                # Two-qubit gate
-                else:
-                    mat = cnot_four_qubits(i.wires.tolist())
-                    mats.append(mat)
+            mats = [op.matrix(wire_order=list(range(4))) for op in decomp]
```
[the remark this answers is not in the document]
Comment (dwierichs) [reply tests/ops/qubit/test_qchem_ops.py:428]: I'm not entirely sure I understand the problem, sorry.
What I _think_ this loop is doing: 
1. For each decomposition from `[decomp1, decomp2]`, create the operation matrices.
2. Multiply the matrices together.
3. Compute the expected matrix using the `gate_data` function `DoubleExcitation` and check that the matrix from the decomposition matches the expectation.

[the remark this answers is not in the document]
Comment (timmysilv) [reply tests/ops/qubit/test_qchem_ops.py:428]: sorry, the GitHub diff made me lose track of indents! I see what you're up to, it looks good to me 👌

**Review by timmysilv — APPROVED** [review]

**Pull request merged**